src/integrationTest/java/com/mongodb/kafka/connect/MongoSourceConnectorTest.java

```diff
@@ -134,10 +141,38 @@ void testSourceLoadsDataFromDatabase() {
         }
     }
 
+    @Test
+    @DisplayName("Ensure source can handle non existent database and survive dropping")
+    void testSourceCanHandleNonExistentDatabaseAndSurviveDropping() throws InterruptedException {
+        try (KafkaConsumer<?, ?> consumer = createConsumer()) {
+            Pattern pattern = Pattern.compile(format("^%s.*", getDatabaseName()));
+            consumer.subscribe(pattern);
+
+            MongoDatabase db = getDatabaseWithPostfix();
+            MongoCollection<Document> coll = db.getCollection("coll");
+
+            Properties sourceProperties = new Properties();
+            sourceProperties.put(MongoSourceConfig.DATABASE_CONFIG, db.getName());
+            addSourceConnector(sourceProperties);
+
+            Thread.sleep(5000);
+            assertProduced(0, coll);
+
+            insertMany(rangeClosed(1, 100), coll);
+            assertProduced(100, coll);
+
+            coll.drop();
```

Comment:
Can a database drop be survived?  IIRC you mentioned something about that last week.

Reply:
Yes  - updating MongoDB seemed to fix it.

Reply:
Should this be a db.drop() then?

Reply:
Good catch. I had to put a sleep in as the collection drop event happens first then the invalidate event some time after.

Reply:
Unfortunate we need the sleep.  I suppose if it causes us problems we can come up with something clever to replace it.